src/lib/utils/mem_ops.h

```diff
@@ -44,15 +47,39 @@ namespace Botan {
 */
 BOTAN_PUBLIC_API(2, 0) void secure_scrub_memory(void* ptr, size_t n);
 
+/**
+* Scrub memory contents in a way that a compiler should not elide,
+* using some system specific technique. Note that this function might
+* not zero the memory.
+*
+* @param data  the data region to be scrubbed
+*/
+void secure_scrub_memory(ranges::contiguous_output_range auto&& data) {
+   secure_scrub_memory(std::ranges::data(data), ranges::size_bytes(data));
+}
+
+#if !defined(BOTAN_IS_BEGIN_BUILT)
+
 /**
 * Memory comparison, input insensitive
 * @param x a pointer to an array
 * @param y a pointer to another array
 * @param len the number of Ts in x and y
 * @return 0xFF iff x[i] == y[i] forall i in [0...n) or 0x00 otherwise
 */
+BOTAN_DEPRECATED("This function is deprecated, use constant_time_compare()")
 BOTAN_PUBLIC_API(2, 9) uint8_t ct_compare_u8(const uint8_t x[], const uint8_t y[], size_t len);
 
+#endif
+
+/**
+ * Memory comparison, input insensitive
+ * @param x a range of bytes
+ * @param y another range of bytes
+ * @return true iff x and y have equal lengths and x[i] == y[i] forall i in [0...n)
+ */
+BOTAN_PUBLIC_API(3, 3) bool constant_time_compare(std::span<const uint8_t> x, std::span<const uint8_t> y);
```

Comment:
Do you think we want to move this to ct_utils.h?

Reply:
Probably. This PR has more than enough complexity, though. So, I suggest we do that as a follow up.

Reply:
ct_utils is our internal header while `constant_time_compare` is *intended* for library users since this comes up for application also.

All internal uses should use `CT::Mask` instead - I have a WIP branch on this somewhere.